utils/markdown/renderer.tsx

```diff
@@ -56,99 +58,105 @@ export default class Renderer extends marked.Renderer {
+        return oneLineTrim`<div class="${className}">
+                ${header}
+                <code class="hljs">
+                    ${searchedContent}
+                    ${content}
+                </code>
+            </div>`;
     }
 
-    codespan(text) {
+    public codespan(text: string) {
         let output = text;
 
         if (this.formattingOptions.searchPatterns) {
             const tokens = new Map();
-            output = TextFormatting.highlightSearchTerms(output, tokens, this.formattingOptions.searchPatterns);
+            output = TextFormatting.highlightSearchTerms(
+                output,
+                tokens,
+                this.formattingOptions.searchPatterns
+            );
             output = TextFormatting.replaceTokens(output, tokens);
         }
 
-        return (
-            '<span class="codespan__pre-wrap">' +
-                '<code>' +
-                    output +
-                '</code>' +
-            '</span>'
-        );
+        return oneLineTrim`<span class="codespan__pre-wrap">
+                <code>
+                    ${output}
+                </code>
+            </span>`;
     }
 
-    br() {
+    public br() {
         if (this.formattingOptions.singleline) {
             return ' ';
         }
 
         return super.br();
     }
 
-    image(href, title, text) {
+    public image(href: string, title: string, text: string) {
         let src = href;
-        let dimensions = [];
+        let dimensions: string[] = [];
         const parts = href.split(' ');
         if (parts.length > 1) {
             const lastPart = parts.pop();
             src = parts.join(' ');
-            if (lastPart[0] === '=') {
+            if (lastPart && lastPart[0] === '=') {
                 dimensions = lastPart.substr(1).split('x');
                 if (dimensions.length === 2 && dimensions[1] === '') {
                     dimensions[1] = 'auto';
                 }
             }
         }
         src = PostUtils.getImageSrc(src, this.formattingOptions.proxyImages);
-        let out = '<img src="' + src + '" alt="' + text + '"';
+        let out = `<img src="${src}" alt="${text}"`;
         if (title) {
-            out += ' title="' + title + '"';
+            out += ` title="${title}"`;
         }
         if (dimensions.length > 0) {
-            out += ' width="' + dimensions[0] + '"';
+            out += ` width="${dimensions[0]}"`;
         }
         if (dimensions.length > 1) {
-            out += ' height="' + dimensions[1] + '"';
+            out += ` height="${dimensions[1]}"`;
         }
         out += ' class="markdown-inline-img"';
         out += this.options.xhtml ? '/>' : '>';
         return out;
     }
 
-    heading(text, level) {
+    public heading(text: string, level: number) {
         return `<h${level} class="markdown__heading">${text}</h${level}>`;
     }
 
-    link(href, title, text, isUrl) {
+    public link(href: string, title: string, text: string) {
```

Comment:
Instead of removing the extra parameter and using `arguments` below, I think it might be possible to provide a default value for `isUrl` to have this pass the type check

Reply:
`link` method's signature on parent class has only 3 params. This is @devinbinnie 's suggestion, or we can patch `@types/marked` to include the forth param `isUrl` or add an overloaded type, your call.

Reply:
I just tested locally, and this seems to work without using `arguments` below
```javascript
public link(href: string, title: string, text: string, isUrl: boolean = false) {}
```

Could you change this to that?

Reply:
I wasn't aware of the solution @hmhealey presents, you can use that solution. My apologies for the confusion.

Reply:
@devinbinnie 
The proposed solution of hmhealey throws a lint error, either we patch the `Renderer` type by adding an overloaded `link` and `listitem` method or use your solution.
```
Renderer.link(href: string, title: string, text: string, isUrl: boolean): string
Property 'link' in type 'Renderer' is not assignable to the same property in base type 'Renderer'.
  Type '(href: string, title: string, text: string, isUrl: boolean) => string' is not assignable to type '(href: string, title: string, text: string) => string'
```

```
(method) Renderer.listitem(text: string, bullet: string): string
Property 'listitem' in type 'Renderer' is not assignable to the same property in base type 'Renderer'.
  Type '(text: string, bullet: string) => string' is not assignable to type '(text: string) => string'.
```

Reply:
That's strange. This seems to be working for me. Here are the changes I made: https://github.com/mattermost/mattermost-webapp/commit/fdeeb121ec3c028d66ac8472f268ac7690193a3c